src/App.tsx

```diff
@@ -1,156 +1,237 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+/* eslint-disable no-param-reassign */
+import classNames from 'classnames';
+import React, { useCallback, useMemo, useState } from 'react';
+
+type Todo = {
+  id: number;
+  title: string;
+  isEditing: boolean;
+};
+
+type Filter = 'All' | 'Completed' | 'Active';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [completedTodos, setCompletedTodos] = useState<Todo[]>([]);
+  const [inputValue, setInputValue] = useState('');
+  const [filter, setFilter] = useState<Filter>('All');
+  const [isVisible, setIsVisible] = useState(true);
+  const [isEditing, setIsEditing] = useState(false);
+  const [editedValue, setEditedValue] = useState('');
+
+  const createTodo = useCallback((title: string) => {
+    return {
+      id: Math.round(Math.random() * 1000) / 1000,
```

Comment:
it can generate not unique ID. It's better to just add numbers starting from `0` or `1`. Define current biggest ID number in the array and add 1 - that will be the ID for the new todo